apps/www/lib/analytics/use-track-page-view.ts

```diff
@@ -0,0 +1,69 @@
+"use-client"
+
+import { useEffect, useMemo, useRef, useState } from "react"
```

Comment:
Should we add a unit test for this? Maybe a mock setup that would test:Event firing on URL change the payload shape and debouncing/deduping repeated events?

Reply:
Yeah, could do